### PR TITLE
fix: Elasticsearch DB Head request fails on some ES instances when used on /test API

### PR DESCRIPTION
**Pull request by sondermanish**
## Summary
	
         * Pom.xml --> ES- Restclient version to 7.17.5 from 7.9.2
	* ElasticSearchPlugin.java ---> changed the testDatasource method to expect a 200 instead of a 200 or 404 HTTP status code

This commit fixes a high-priority bug with github issue  https://github.com/appsmithorg/appsmith/issues/14909
	* it can now negotiate around a 403 forbidden error as it does try to send head only to user provided  URI

This commit has been tested:
	* manually for CRUD at local with ES version 7.9.2
	* manually for CR at user provide AWS instance running 7.9.3 on docker
	* JUnit testcases

## Fixes

This commit fixes Head request failure when running `/test` API on  some ES instances which throws 403 forbidden. 

## How Has This Been Tested?

The code has been manually tested for the create read delete and update use cases manually on locally hosted ES instance and server hosted ES instance

## Checklist:

- [✅] My code follows the style guidelines of this project
- [✅] I have performed a self-review of my own code
- [✅] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [✅] I have added tests that prove my fix is effective or that my feature works
- [✅] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-plugins/elasticSearchPlugin/src/test/java/com/external/plugins/ElasticSearchPluginTest.java [hunk 1]
```diff
@@ -300,8 +328,48 @@ public void itShouldTestDatasourceWithInvalidEndpoint() {
     public void itShouldTestDatasource() {
         StepVerifier.create(pluginExecutor.testDatasource(dsConfig))
                 .assertNext(result -> {
+                    for (String message: result.getInvalids())System.out.println("------- meow" +message);
```
Comment (sumitsum) [inline app/server/appsmith-plugins/elasticSearchPlugin/src/test/java/com/external/plugins/ElasticSearchPluginTest.java:331]: Should this print statement be removed ?

**Review by sumitsum — CHANGES_REQUESTED** [review]

File: app/server/appsmith-plugins/elasticSearchPlugin/src/test/java/com/external/plugins/ElasticSearchPluginTest.java (hunk 1 above)
Comment (sondermanish) [reply app/server/appsmith-plugins/elasticSearchPlugin/src/test/java/com/external/plugins/ElasticSearchPluginTest.java]: removed the debug comments.

File: app/server/appsmith-plugins/elasticSearchPlugin/src/main/java/com/external/plugins/ElasticSearchPlugin.java [hunk 2]
```diff
@@ -61,6 +63,15 @@ public static class ElasticSearchPluginExecutor implements PluginExecutor<RestCl
 
         private final Scheduler scheduler = Schedulers.elastic();
 
+        public static final String esDatasourceNotFoundMessage = "404 Error: The Page you are tyring to access does not exist";
```
[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-plugins/elasticSearchPlugin/src/main/java/com/external/plugins/ElasticSearchPlugin.java]: modified as suggested.

**Review by sumitsum — APPROVED** [review]

File: app/server/appsmith-plugins/elasticSearchPlugin/src/test/java/com/external/plugins/ElasticSearchPluginTest.java [hunk 3]
```diff
@@ -1,14 +1,17 @@
 package com.external.plugins;
 
-import com.appsmith.external.models.ActionConfiguration;
-import com.appsmith.external.models.ActionExecutionResult;
-import com.appsmith.external.models.DatasourceConfiguration;
-import com.appsmith.external.models.Endpoint;
-import com.appsmith.external.models.RequestParamDTO;
+import com.appsmith.external.constants.Authentication;
+import com.appsmith.external.models.*;
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/elasticSearchPlugin/src/test/java/com/external/plugins/ElasticSearchPluginTest.java:4]: Please remove wildcard imports

**Comment by Aishwarya-U-R** [comment]
Adding 'Test Plan Approved' since cypress is NA for this DS.

File: app/server/appsmith-plugins/elasticSearchPlugin/src/test/java/com/external/plugins/ElasticSearchPluginTest.java (hunk 3 above)
Comment (sondermanish) [reply app/server/appsmith-plugins/elasticSearchPlugin/src/test/java/com/external/plugins/ElasticSearchPluginTest.java]: Changed as suggested.

**Review by sumitsum — APPROVED** [review]

**Pull request merged**